plugins/filter/agent.py

```diff
@@ -24,6 +26,56 @@ def to_agent_version(data):
         return "latest"
 
 
+def to_agent_version_pinned(data):
+    """ Returns True when the agent version to install is pinned, False otherwise
+    """
+    v = to_agent_version(data)
+    if v and v != '' and v != "latest":
+        return True
+    return False
+
+def to_agent_packages(data):
+    """ Returns the agent packages to install
+    """
+    pinned = to_agent_version_pinned(data)
+    older = False
+    all_packages = {
+        "draios-agent": False,
+        "draios-agent-slim": False,
+        "draios-agent-kmodule": False,
+        "draios-agent-legacy-ebpf": False,
+    }
+    if pinned:
+        version = to_agent_version(data)
+        minver = LooseVersion("1.0.0")
+        maxver = LooseVersion("13.1.0")
+        older = minver < LooseVersion(version) < maxver
+    if older:
+        all_packages ["draios-agent"] = True
+    else:
+        dt = to_agent_driver_type(data)
+        if dt == "universal_ebpf":
+            all_packages ["draios-agent-slim"] = True
+        elif dt == "legacy_ebpf":
+            all_packages ["draios-agent-slim"] = True
+            all_packages ["draios-agent-legacy-ebpf"] = True
+        else:
+            all_packages ["draios-agent"] = True
+            all_packages ["draios-agent-slim"] = True
+            all_packages ["draios-agent-kmodule"] = True
```

Comment:
At least to me, keying the package map off of the driver type makes things a bit more straight forward. As an example:
```python
package_map = {
  "universal_ebpf": ["draios-agent-slim"],
  "legacy_ebpf": ["draios-agent-slim", "draios-agent-legacy-ebpf"],
  "kmod": ["draios-agent", "draios-agent-slim", "draios'agent-kmodule"]
}
...
if older:
  return ["draios-agent"]
return package_map[to_agent_driver_type(data)]
```
In this way we don't have to manually "flip the switch" for packages to install for a given driver and also there is a clear mapping between driver type and package.

The `to_agent_install_packages()` and `to_agent_uninstall_packages()` methods can be simplified to something like this as well:

**to_agent_uninstall_packages()**
```python
# hacky global. could make better.
AGENT_PACKAGES = ["draios-agent", "draios-agent-legacy-ebpf", "draios-agent-slim", "draios-agent-kmodule"]
...
def to_agent_uninstall_packages(data):
  return [p for p in AGENT_PACKAGES if p not in to_agent_packages(data)]
```

**to_agent_install_packages(data):**
```python
def to_agent_install_packages(data):
  return to_agent_packages(data)
```